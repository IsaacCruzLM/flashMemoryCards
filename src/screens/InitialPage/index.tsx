import * as React from 'react';
import {Text} from 'react-native';

import LinearGradientView from '../../components/LinearGradientView';

const InitialPage = () => {
  return (
    <LinearGradientView>
      <Text>TExto qualquer</Text>
      <Text>TExto qualquer 2</Text>
    </LinearGradientView>
  );
};

export default InitialPage;
