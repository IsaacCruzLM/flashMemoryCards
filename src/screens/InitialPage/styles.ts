import {StyleSheet} from 'react-native';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    width: '100%',
    paddingVertical: 100,
    paddingHorizontal: 24,
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  buttonContainer: {
    width: '100%',
  },
});

export default styles;
