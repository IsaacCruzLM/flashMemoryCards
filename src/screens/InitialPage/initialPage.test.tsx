/**
 * @format
 */
import React from 'react';
import 'react-native';
import InitialPage from './index';
import renderer from 'react-test-renderer';

jest.mock('../../navigation/NavigationService', () => ({
  navigate: jest.fn(),
}));

jest.mock('../../assets/logo.svg', () => 'LogoSvg');

jest.mock('../../components/Button', () => {
  return {
    __esModule: true,
    default: ({label}: {onPress: () => void; label: string}) =>
      `Button-${label}`,
  };
});

describe('InitialPage', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const tree = renderer.create(<InitialPage />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('navigates to Tutorial screen when button is pressed', () => {
    const NavigationService = require('../../navigation/NavigationService');
    const component = renderer.create(<InitialPage />);

    // Find the Button component
    const instance = component.root;
    const button = instance.findByProps({label: 'Entrar'});

    // Trigger the onPress event
    button.props.onPress();

    // Verify navigation
    expect(NavigationService.navigate).toHaveBeenCalledWith('Tutorial');
  });

  it('initializes animations on mount', () => {
    const reanimated = require('react-native-reanimated');
    renderer.create(<InitialPage />);

    // Verify that animations are initialized
    expect(reanimated.useSharedValue).toHaveBeenCalledWith(0);
    expect(reanimated.useSharedValue).toHaveBeenCalledWith(30);
    expect(reanimated.withTiming).toHaveBeenCalled();
    expect(reanimated.withRepeat).toHaveBeenCalled();
  });
});
