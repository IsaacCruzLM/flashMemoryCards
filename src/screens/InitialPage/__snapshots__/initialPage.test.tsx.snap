// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InitialPage renders correctly 1`] = `
<View
  colors={
    Array [
      "rgba(34, 39, 46, 0.7)",
      "rgba(34, 39, 46, 1)",
    ]
  }
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "space-between",
        "paddingHorizontal": 24,
        "paddingVertical": 100,
        "width": "100%",
      }
    }
  >
    <View
      style={[Function]}
    >
      <LogoSvg
        fill="#ffffff"
        height={78}
        width={93}
      />
    </View>
    <View
      style={
        Object {
          "width": "100%",
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 20,
              "fontWeight": "700",
              "marginBottom": 30,
              "textAlign": "center",
            },
            [Function],
          ]
        }
      >
        Bem vindo(a) ao
Flash Memory
Cards
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#FFFFFF",
              "fontFamily": "System",
              "fontSize": 14,
              "fontWeight": "500",
              "marginBottom": 70,
              "textAlign": "center",
            },
            [Function],
          ]
        }
      >
        Seja relembrado de suas anotações, para que o conhecimento seja realmente fixado!
      </Text>
      <View
        style={[Function]}
      >
        Button-Entrar
      </View>
    </View>
  </View>
</View>
`;
